### PR TITLE
chore(documentation): Updates usage of all widgets. Fixes #396

### LINKED ISSUE #396: Check usages of all widgets
Many are inconsistent.

Please check that all of the "options" are stated as "options.optionName" in the jsdoc of widgets, for them to render well in the website documentation.

### PR DESCRIPTION
**Pull request by bobylito**
Also:
 - fixes tests for bad usage
 - makes parameters consistent

### CODE REVIEW

**Comment by vvo** [comment]
> use the consistent attributeName for parameter instead of facetName

Not sure about this one, we need to discuss it first right?

`facetName` was chosen for all widgets doing **faceting** while attributeName is more for numeric filters. It's only a way to map our users to the information they will find in the algolia dashboard (refinement list requirement: use attributes for faceting).

Still, we can change it maybe but it needs to be in another PR I think.

File: mocha.opts [hunk 1]
```diff
@@ -1,2 +1,2 @@
 --compilers js:babel/register
-@(components|widgets|lib|decorators)/**/*-test.js
```
Comment (vvo) [inline mocha.opts]: why would it not be mandatory? It was mapped on what most of the react community does (https://github.com/rackt/react-router/tree/master/modules/__tests__)

Comment (vvo) [inline mocha.opts]: We can switch to no -test.js but then it should go in another PR (moving all -test.js), for consistency

**Comment by bobylito** [comment]
Oh sorry I remembered a discussion about that, and acted as I understood it. No biggie will fix that :)

File: widgets/price-ranges/__tests__/price-ranges-test.js [hunk 2]
```diff
@@ -12,16 +12,17 @@ import priceRanges from '../price-ranges';
 import generateRanges from '../generate-ranges';
 import PriceRanges from '../../../components/PriceRanges/PriceRanges';
 
-describe('priceRanges call', () => {
+describe('priceRange call', () => {
```
Comment (vvo) [inline widgets/price-ranges/__tests__/price-ranges-test.js]: widget is named priceRanges, why priceRange here?

File: mocha.opts (hunk 1 above)
Comment (bobylito) [inline mocha.opts]: Meh ok will remove that :)

**Comment by vvo** [comment]
> Oh sorry I remembered a discussion about that, and acted as I understood it. No biggie will fix that :)

It is still a discussion we can have, maybe that's not good what we did but should be ok for a first release I guess

File: widgets/price-ranges/__tests__/price-ranges-test.js (hunk 2 above)
Comment (bobylito) [inline widgets/price-ranges/__tests__/price-ranges-test.js]: Bad rebase :)

**Comment by bobylito** [comment]
I have mixed feelings about the attributeName stuff. I acted on my own here, so I guess I should revert that anyway.

File: mocha.opts (hunk 1 above)
Comment (bobylito) [inline mocha.opts]: I didn't realise that we didn't enforce __tests__ :/

**Comment by bobylito** [comment]
@vvo are we good to go now?

**Comment by vvo** [comment]
very nice to have this for 1.0.0! I would love a lib that give me so much detail about what I did wrong without me having to go to the doc, again

**Comment by bobylito** [comment]
#DX :)

**Comment by pixelastic** [comment]
Lovely, thanks a lot for that
